src/components/Dialogs/DataExport/DataExport.spec.js

```diff
@@ -0,0 +1,256 @@
+import React from 'react';
+import { DataExport } from './DataExport';
+import * as utils from '../../../utils';
+import {
+  testRender as render,
+  screen,
+  fireEvent,
+} from '../../../testUtils/test-utils';
+import { merge } from '../../../testUtils/functionHelpers';
+import { defaultAggs } from '../../../reducers/aggs/aggs';
+import { defaultQuery } from '../../../reducers/query/query';
+import * as viewActions from '../../../actions/view';
+import { MODAL_TYPE_EXPORT_CONFIRMATION } from '../../../constants';
+
+describe('DataExport', () => {
+  const originalClipboard = { ...global.navigator.clipboard };
+
+  const renderComponent = (newAggsState, newQueryState) => {
+    const mockClipboard = {
+      writeText: jest.fn(),
+    };
+    global.navigator.clipboard = mockClipboard;
+
+    merge(newAggsState, defaultAggs);
+    merge(newQueryState, defaultQuery);
+    const data = {
+      aggs: newAggsState,
+      query: newQueryState,
+    };
+    render(<DataExport />, { preloadedState: data });
+  };
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    global.navigator.clipboard = originalClipboard;
+  });
+
+  it('renders default state without crashing', async () => {
+    const hideModalSpy = jest
+      .spyOn(viewActions, 'hideModal')
+      .mockImplementation(() => jest.fn());
+    renderComponent({}, {});
+    expect(screen.getByText('Export complaints')).toBeInTheDocument();
+
+    // hide dataset buttons when no filters selected
+    expect(
+      screen.queryByText(/Select which complaints you'd like to export/)
+    ).toBeNull();
+    expect(
+      screen.getByText(
+        'Link to your complaint search results for future reference'
+      )
+    ).toBeInTheDocument();
+
+    expect(
+      screen.getByRole('button', { name: 'Start export' })
+    ).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+    const buttonCopy = screen.getByRole('button', { name: 'Copy' });
+    expect(buttonCopy).toBeInTheDocument();
+    expect(buttonCopy).toHaveClass('a-btn__secondary');
+    fireEvent.click(buttonCopy);
+    await (() => {
```

Comment:
This should be `await waitFor( () => {...} ); ` The `await` by itself will pass everything

Reply:
awesome, thanks!

Reply:
You're right.  By adding waitFor and also following the eslint nag of not combining the await conditions, an error was caught.